assembl/views/auth/views.py

```diff
@@ -1126,53 +1126,54 @@ def send_change_password_email(
     mailer = get_mailer(request)
     localizer = request.localizer
     route_maker = create_get_route(request, discussion)
-    data = dict(
-        assembl="Assembl", name=profile.name,
-        confirm_url=get_global_base_url() + route_maker(
-            'welcome' if welcome else 'do_password_change',
-            token=password_change_token(profile)))
-    sender_email = config.get('assembl.admin_email')
-    if discussion:
-        data.update(dict(
-            discussion_topic=discussion.topic,
-            discussion_url=discussion.get_url()))
-        sender_name = sender_name or discussion.topic
-    if sender_name:
-        sender_name = normalize_email_name(sender_name)
-        sender = '"%s" <%s>' % (sender_name, sender_email)
-        sender_name = Header(sender_name, 'utf-8').encode()
-        if len(sender) > 255:
+    if profile:
```

Comment:
I see what you're trying to do, and it's a good idea to be more defensive in this function against the profile not existing, etc. However, perhaps what you really need is another function, one that either sends the email, or returns an `HTTPOk`, according to the error that is passed into it. The function, if the profile doesn't exist, will do nothing, which is dangerous if a future code change breaks sending the profile, and then nothing will happen, and tracking emails being sent/not-sent is not trivial at all.